src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/SearchNavSideMenuComponentObject.java

```diff
@@ -0,0 +1,153 @@
+package com.wikia.webdriver.pageobjectsfactory.componentobject.mercury;
+
+import com.wikia.webdriver.common.contentpatterns.MercuryContent;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import org.openqa.selenium.WebDriver;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.mercury.MercuryBasePageObject;
+
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+
+/*
+* @author: Rodrigo Gomez, Łukasz Nowak, Tomasz Napieralski
+* */
+public class SearchNavSideMenuComponentObject extends MercuryBasePageObject{
+
+	@FindBy(css = ".ember-text-field")
+	private WebElement searchInput;
+	@FindBy(css = ".local-nav-menu > li")
+	private WebElement firstLevelMenuPositions;
+	@FindBy(css = ".local-nav-menu > title")
+	private WebElement menuTitle;
+	@FindBy(css = "li[class='needsclick']")
+	private WebElement needsClickMenuPositions;
+	@FindBy(css = ".cancel")
+	private WebElement cancelSearchCaption;
+	@FindBy(css = ".local-wikia-search > li > a")
+	private WebElement searchResults;
+	@FindBy(css = "li[class='back']")
+	private WebElement menuBackButton;
+	@FindBy(css = ".icon chevron")
+	private WebElement needsClickChevrons;
+	@FindBy(css = ".local-wikia-search a")
+	private List<WebElement> searchSuggestions;
+	@FindBy(css = ".local-wikia-search span")
+	private WebElement searchAlerts;
+	@FindBy(css = ".search")
+	private WebElement searchView;
+	@FindBy(css = "div[class='drawer']")
+	private WebElement menuView;
+	@FindBy(css = ".local-nav-menu > li > div")
+	private List<WebElement> chevrons;
+	@FindBy(css = ".local-nav-menu > li > a")
+	private List<WebElement> noChevrons;
+	@FindBy(css= ".back")
+	private WebElement backChevron;
+	@FindBy(css = ".overlay")
+	private WebElement overlay;
+
+	public SearchNavSideMenuComponentObject(WebDriver driver) {
+		super(driver);
+	}
+
+	public void clickSearchField() {
+		waitForElementByElement(searchInput);
+		searchInput.click();
+		PageObjectLogging.log("clickSearchField", "Search field was clicked", true);
+	}
+
+	public MercuryBasePageObject clickSearchSuggestion(int index) {
+		waitForElementByElement(searchSuggestions.get(index));
+		searchSuggestions.get(index).click();
+		return new MercuryBasePageObject(driver);
+	}
+
+	public void clickCancelButton() {
+		waitForElementVisibleByElement(cancelSearchCaption);
+		cancelSearchCaption.click();
+	}
+
+	public String getSearchResultHref(int searchPosition) {
+		waitForElementVisibleByElement(searchSuggestions.get(searchPosition));
+		return searchSuggestions.get(searchPosition).getAttribute("href");
+	}
+
+	public void typeInSearchField(String content, int length) {
+		waitForElementVisibleByElement(searchInput);
+		searchInput.sendKeys(content.substring(0, length));
```

Comment:
move the logic for trimming the 'content' outside of the fucntion. 
The intuitive understanding of this function is that it will put 'content' into search field.
So the method is: public void typeInSearchField(String content), and you trim content in the test